### DISCUSSION TITLE
Research and reflect: Would Gitbook improve the user experience for docs contributions? What else could we use?

### DISCUSSION THREAD

**Discussion by stichbury** [kedro-org/kedro#3749] [Idea]
I'd like to make it easier for contributors to make changes to documentation. Hell, I'd like to make it easier for the team to make documentation changes, and I'd like to make it easier for myself too.

Right now, if you spot, say, a typo on a docs page you have to look at the page name in the URL, fork or clone the Kedro `main` branch, and branch off it, navigate to the file in `docs/source` open and edit it. Then sign your commit, push to the remote and wait for the whining to start from the CI. Fix anything you've broken, like the whitespace or a DCO, and re-try. Etc. Until you get the green tick of acceptance from CI and can initiate review and merge.

I'd love to see this made simpler for people that don't do it all the time. I recently saw a [Gitbook](https://www.gitbook.com/) demo and it looks much better for contributors, while still giving power to manage PRs and veto changes to the code owners.

I wonder if this is a possible way forward for Kedro docs, at least markdown docs, but I appreciate it's not the only option.

So this is a ticket to generally look at how we could streamline the process of docs contributions and potentially this bleeds into a more holistic overhaul of the docs toolchain. However, it's not a priority until other milestones are addressed, such as #2600 for subprojects and #2932 for Algolia search improvements.

**Comment by astrojuanlu** [comment]
I think if we could get a DCO exception for docs + Add a "Edit on GitHub" button that actually took users to the `.md` file in question that would already be a massive improvement 👍🏽

**Comment by stichbury** [comment]
Would be good to get some steer from LF on how other projects manage the DCO for docs contribs, as it definitely slows down the process. Particularly if they're not that familiar with what we expect, so it takes a fair bit of back and forth and may even put some people off doing another PR. I like the idea of just hit a button, fix a line of text, and commit it, if that's any possibility without massive amounts of changes to our tooling.

**Comment by astrojuanlu** [comment]
We agreed to reach out to LF to have some insight about this.

In addition, looks like some IDEs like VSCode allow configuring this automatically, see https://github.com/microsoft/vscode/issues/83096#issuecomment-545350047

![image](https://github.com/kedro-org/kedro/assets/316517/c542f4bc-2e4e-40fe-ac05-bd6591b83309)

**Comment by astrojuanlu** [comment]
Also, it's unclear to me if we have this enabled, but there's definitely a way to ease the contribution process for web commits https://github.blog/changelog/2022-06-08-admins-can-require-sign-off-on-web-based-commits/


![signoff_highlight](https://github.com/kedro-org/kedro/assets/316517/56495a32-0895-4a3a-8c92-70bef64408fc)

**Comment by astrojuanlu** [comment]
Yeah just spotted this myself when tweaking a text file from the GitHub URL 
![Screenshot 2023-10-06 at 10-28-40 Editing good-first-issue_pyproject toml at patch-2 · astrojuanlu_good-first-issue](https://github.com/kedro-org/kedro/assets/316517/2f7e00fd-355f-4654-9240-fa4e0692fa3e)

**Comment by astrojuanlu** [comment]
Reached out to LF AI & Data.

**Comment by astrojuanlu** [comment]
I confirm that, after enabling web signoff for kedro-plugins, now I see this on our own repo

![Screenshot 2023-10-17 at 08-35-18 Editing kedro-plugins_kedro-airflow_README md at main · kedro-org_kedro-plugins](https://github.com/kedro-org/kedro/assets/316517/69b6ff7b-d532-4f8d-8276-cb62da09448f)

On the other hand, we got permission from LF AI & Data to disable the DCO bot for non-code repos.

**Comment by stichbury** [comment]
The range of responses from LF about whether we could disable DCO on our framework repo was somewhat unsatisfactory.  But since the final response was "no", I guess that's the one to take. 

DCO doesn't just affect docs contributions, I don't see the upheaval of moving docs into a separate repo just to disable it as paying off across the board for contributors. If we were to consider such a step, I'd also suggest we review a bunch of other changes to improve docs contributions, such as a switch to Gitbook, which was the purpose of this ticket initially.

So that was a useful detour but ultimately this is still to be addressed and still a lower priority than other changes. I'll leave it open pending a period for reflecting on docs tooling, which won't be until 2024 at the earliest.

**Comment by astrojuanlu** [comment]
FTR, I've enabled the "sign-off web commits" on _this_ repo too. Let's see if that improves things.

**Comment by astrojuanlu** [comment]
De-prioritizing this for the time being.
